### PR TITLE
ARROW-246: [Java] UnionVector doesn't call allocateNew() when creating it's vect…

### PR DESCRIPTION
**Pull request by adeneche**
…orType

### CODE REVIEW

**Comment by adeneche** [comment]
@StevenMPhillips can you please review ?

File: java/vector/src/test/java/org/apache/arrow/vector/TestValueVector.java [hunk 1]
```diff
@@ -72,6 +73,47 @@ public void terminate() throws Exception {
     allocator.close();
   }
 
+  @Test
+  public void testUnionVector() throws Exception {
```
Comment (julienledem) [inline java/vector/src/test/java/org/apache/arrow/vector/TestValueVector.java]: possibly move this to a TestUnionVector class?

**Comment by julienledem** [comment]
LGTM. Will let @StevenMPhillips confirm.

File: java/vector/src/test/java/org/apache/arrow/vector/TestValueVector.java (hunk 1 above)
Comment (adeneche) [inline java/vector/src/test/java/org/apache/arrow/vector/TestValueVector.java]: sounds good

**Comment by julienledem** [comment]
@adeneche Can you add "ARROW-246: " as a prefix to your PR title? See other PRs

File: java/vector/src/test/java/org/apache/arrow/vector/NonCleanBufferAllocator.java [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+import org.apache.arrow.memory.BufferManager;
+
+import io.netty.buffer.ArrowBuf;
+import io.netty.buffer.ByteBufAllocator;
+
+/**
+ * Wrapper around a buffer delegate that populates any allocated buffer with a constant
+ * value. Useful for testing if value vectors are properly resetting their buffers.
+ */
+public class NonCleanBufferAllocator implements BufferAllocator {
```
Comment (jacques-n) [inline java/vector/src/test/java/org/apache/arrow/vector/NonCleanBufferAllocator.java]: Maybe "DirtyBufferAllocator" ?

**Comment by StevenMPhillips** [comment]
+1

**Pull request closed without merging**